source/collections/Vector2D.ooc

```diff
@@ -14,40 +14,92 @@
 * You should have received a copy of the GNU Lesser General Public License
 * along with this software. If not, see <http://www.gnu.org/licenses/>.
 */
+import math
 
 Vector2D: class <T> {
 	_backend: T*
-	_rowCapacity: Int
-	_columnCapacity: Int
-	rowCapacity ::= this _rowCapacity
-	columnCapacity ::= this _columnCapacity
-	_freeContent: Bool
-
-	init: func ~preallocated (=_backend, =_rowCapacity, =_columnCapacity, freeContent := true)
-	init: func (=_rowCapacity, =_columnCapacity, freeContent := true) {
-		this _freeContent = freeContent
-		this _allocate(rowCapacity, columnCapacity)
-		memset(this _backend, 0, rowCapacity * columnCapacity * T size)
+	_rowCount: Int
+	_columnCount: Int
+	rowCount ::= this _rowCount
+	columnCount ::= this _columnCount
+
+	init: func ~preallocated (=_backend, =_rowCount, =_columnCount)
+	init: func (=_rowCount, =_columnCount) {
+		this _allocate(this rowCount, this columnCount)
+		memset(this _backend, 0, this rowCount * this columnCount * T size)
 	}
 	_allocate: func (rows, columns: Int) {
 		this _backend = gc_realloc(this _backend, rows * columns * T size)
 	}
 	free: override func {
 		gc_free(this _backend)
+		super()
 	}
-	_elementPosition: func (row, column: Int, columnCount := this columnCapacity) -> Int {
+	_elementPosition: func (row, column: Int, columnCount := this columnCount) -> Int {
 		columnCount * row + column
 	}
+	resize: func (newRowCount, newColumnCount: Int) {
+		if (newColumnCount == this columnCount) {
+			if (newRowCount != this rowCount) {
+				this _allocate(newRowCount, newColumnCount)
+				this _rowCount = newRowCount
+				this _columnCount = newColumnCount
+			}
+		} else {
```

Comment:
That's logically not the same though, is it? This would cause us to jump into the else statement if both `newColumnCount == this columnCount` and `newRowCount == this rowCount` holds true. In this case, we don't want any code to execute.